### PR TITLE
Remove deprecated plugins and code before major release

### PR DESCRIPTION
**Pull request by ivomurrell**
# Description

Now that we're gearing up to do a major release of the `next` branch, let's incorporate some cleanup of old deprecated code which we've kept around to avoid breaking changes. 

# Checklist:

- [x] My branch has been rebased onto the latest commit on main (don't merge main into your branch)
- [x] My commit messages are [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), for example: `feat(circleci): add support for nightly workflows`, `fix: set Heroku app name for staging apps too`

### CODE REVIEW

File: plugins/circleci-deploy/readme.md [hunk 1]
```diff
@@ -35,12 +35,12 @@ See [manually adding jobs to your CircleCI config](https://github.com/Financial-
 
 ## Hooks
```
Comment (apaleslimghost) [inline plugins/circleci-deploy/readme.md:36]: **thought**: we should update these readmes! didn't get caught in #629

File: plugins/heroku/test/promoteStagingToProduction.test.ts [hunk 2]
```diff
@@ -3,9 +3,8 @@ import { promoteStagingToProduction } from '../src/promoteStagingToProduction'
 import heroku from '../src/herokuClient'
 import { gtg } from '../src/gtg'
 import winston, { Logger } from 'winston'
-import { setAppConfigVars } from '../src/setConfigVars'
 
-const logger = (winston as unknown) as Logger
+const logger = winston as unknown as Logger
```
Comment (apaleslimghost) [inline plugins/heroku/test/promoteStagingToProduction.test.ts:7]: **question**: i'm sure this keeps flipflopping, what's causing that?

**Review by apaleslimghost — APPROVED** [review]
_mario voice_ buh-bye!

File: plugins/heroku/test/promoteStagingToProduction.test.ts (hunk 2 above)
Comment (ivomurrell) [reply plugins/heroku/test/promoteStagingToProduction.test.ts:7]: looks like it's just the heroku tests that still have parentheses in this casting at this point, i expect we're just slowly applying prettier throughout the source as we make changes. (i still need to fix this test logging, winston doesn't like you just using the root logger without any transports configured and will print loads of complaints to stderr as soon as you switch off `--silent`)

**Pull request merged**